spec/asyncapi.md

```diff
@@ -529,167 +529,197 @@ defaultContentType: application/json
-          },
-          "signup": {
-            "$ref": "#/components/schemas/signup"
+  "channels": {
+    "userEvents": {
+      "address": "users.{userId}",
+      "addressDelimiter": ".",
+      "description": "This channel is used to exchange messages about user events.",
+      "messages": {
+        "userSignedUp": {
+          "$ref": "#/components/messages/userSignedUp"
+        },
+        "userCompletedOrder": {
+          "$ref": "#/components/messages/userCompletedOrder"
+        }
+      },
+      "parameters": {
+        "userId": {
+          "$ref": "#/components/parameters/userId"
+        }
+      },
+      "servers": ["rabbitmqInProd", "rabbitmqInStaging"],
+      "bindings": {
+        "amqp": {
+          "is": "queue",
+          "queue": {
+            "exclusive": true
           }
         }
-      }
-    }
-  },
-  "bindings": {
-    "amqp": {
-      "is": "queue",
-      "queue": {
-        "exclusive": true
+      },
+      "tags": [{
+        "name": "user",
+        "description": "User-related messages"
+      }],
+      "externalDocs": {
+        "description": "Find more info here",
+        "url": "https://example.com"
       }
     }
   }
 }
 ```
 
 ```yaml
-description: This channel is used to exchange messages about users signing up
-subscribe:
-  summary: A user signed up.
-  message:
-    description: A longer description of the message
-    payload:
-      type: object
-      properties:
-        user:
-          $ref: "#/components/schemas/user"
-        signup:
-          $ref: "#/components/schemas/signup"
-bindings:
-  amqp:
-    is: queue
-    queue:
-      exclusive: true
+channels:
+  userEvents:
+    address: 'users.{userId}'
+    addressDelimiter: '.'
+    description: This channel is used to exchange messages about user events.
+    messages:
+      userSignedUp:
+        $ref: '#/components/messages/userSignedUp'
+      userCompletedOrder:
+        $ref: '#/components/messages/userCompletedOrder'
+    parameters:
+      userId:
+        $ref: '#/components/parameters/userId'
+    servers:
+      - rabbitmqInProd
+      - rabbitmqInStaging
+    bindings:
+      amqp:
+        is: queue
+        queue:
+          exclusive: true
+    tags:
+      - name: user
+        description: User-related messages
+    externalDocs:
+      description: 'Find more info here'
+      url: 'https://example.com'
 ```
 
-Using `oneOf` to specify multiple messages per operation:
 
-```json
-{
-  "subscribe": {
-    "message": {
-      "oneOf": [
-        { "$ref": "#/components/messages/signup" },
-        { "$ref": "#/components/messages/login" }
-      ]
-    }
-  }
-}
-```
 
-```yaml
-subscribe:
-  message:
-    oneOf:
-      - $ref: '#/components/messages/signup'
-      - $ref: '#/components/messages/login'
-```
 
 
-Using explicit by-name references to the servers on which the channel is available:
+#### <a name="channelAddressExpressions"></a>Channel Address Expressions
```

Comment:
I went all-in defining our own templating rules here. I don't know what I was thinking when I defined the channel address as a URL template in version 2.0.0 😅 Have a look: https://www.rfc-editor.org/rfc/rfc6570#section-2.4.2.